src/libstd/ffi/os_str.rs

```diff
@@ -312,12 +312,19 @@ impl Default for OsString {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl Debug for OsString {
-    fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+impl fmt::Debug for OsString {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
         fmt::Debug::fmt(&**self, formatter)
     }
 }
 
+#[stable(feature = "rust1", since = "1.20.0")]
+impl fmt::Display for OsString {
```

Comment:
`Display` for `OsString` seems to be perfectly reasonable for me. You may need to "display" an OS string like you "display" the usual string.

Anywhay, if it should exist, how should I mute `tidy`?

If it shouldn't exist, how should I keep an implementation, but hide it from public API? `#[doc(hidden)]` and `#[unstable(feature = "str_internals", issue = "0")]`?